src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -520,7 +520,7 @@ protected Object clone() throws CloneNotSupportedException {
      * @since 9
      */
     @ForceInline
-    public static void reachabilityFence(Object ref) {
+    public static void reachabilityFence(@Nullable Object ref) {
```

Comment:
What use is there to allow null here? I agree that the documentation says that null is allowed, so no NPE will be thrown. On the other hand, a reachability fence only makes sense for a real object. Doesn't allowing null here simply allow likely buggy code?
Can you add a `CFComment` that adds a rationale for making this parameter nullable?

Reply:
Only the constructor changes were motivated by real code; I just noticed `reachabilityFence` while I was in the area.

This method is rarely called. And, at least inside Google, it's always called with an argument of `this`. So the choice of annotation is probably irrelevant in practice. I've removed `@Nullable` and added a `@CFComment` about this.